src/modules/products/repositories/product.repository.ts

```
@@ -0,0 +1,89 @@
+      const allowedSortFields = ["name", "price", "createdAt"];
+      if (sortBy && allowedSortFields.includes(sortBy)) {
+        queryBuilder.orderBy(`product.${sortBy}`, order === "DESC" ? "DESC" : "ASC");
+      }
+
+   
+      const offset = (page - 1) * limit;
+      queryBuilder.skip(offset).take(limit);
+
+    
+      if (fields) {
+        const selectedFields = fields.split(",").map((field) => `product.${field.trim()}`);
```

ESLint

[error] Replace `.split(",")` with `⏎··········.split(',')⏎··········`

(prettier/prettier)